### PR TITLE
Keymap parsing and keypress handling improvements

### PR DESCRIPTION
**Pull request by bugadani**
This PR:
 - Adds an empty common keymap file, to allow deduplicating existing default keymaps
 - Cleans up the keymap parsing function to use simpler code and avoid several unnecessary allocations.

These changes came up while trying to figure out the cause of #116 and #185 - I didn't make any progress on that front, but these changes might make the parser code slightly easier to understand. Maybe, hopefully :)

### CODE REVIEW

File: defaults/keymaps-common.toml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+command = "right"
+
+[[keymaps]]
+key = "left"
+command = "left"
+
+[[keymaps]]
+key = "up"
+command = "up"
+when = "!list_focus"
```
Comment (bugadani) [inline defaults/keymaps-common.toml:110]: `up` and `down` only had the `when = "!list_focus"` in the macOS keymap. I don't know which version of this is the intended, so I picked the macOS one and added the `when`.

**Comment by bugadani** [comment]
I put this back to draft because I want to continue refactoring to avoid parsing the TOML strings twice.

File: lapce-ui/src/app.rs [hunk 2]
```diff
@@ -26,8 +26,8 @@ pub fn lanuch() {
                 message
             ))
         })
-        .level(log::LevelFilter::Off)
-        .level_for("piet_wgpu", log::LevelFilter::Info);
+        .level(log::LevelFilter::Info)
```
Comment (bugadani) [inline lapce-ui/src/app.rs:29]: Note to self: Don't forget to revert this change

**Comment by dzhou121** [comment]
Can I suggest to keep a PR small so that it’s easier to merge?

**Comment by bugadani** [comment]
I can split it up and submit it in parts once I'm done. Parts of it are small (like the `Modes`) change, but splitting up into multiple files will naturally stay rather large.

File: lapce-data/src/keypress/mod.rs [hunk 3]
```diff
@@ -370,17 +370,14 @@ impl KeyPressData {
 
         if mode != Mode::Insert
```
Comment (bugadani) [inline lapce-data/src/keypress/mod.rs:371]: Checking `mode` here looks redundant. `handle_count` already checks these conditions, but here the code is checking the original focus, not the potentially changed one.

@dzhou121 if the focus changes, why does this code check the original focus' mode?

File: lapce-data/src/keypress/mod.rs [hunk 4]
```diff
@@ -352,10 +352,8 @@ impl KeyPressData {
             KeymapMatch::None => {
                 self.pending_keypress.clear();
                 if focus.get_mode() == Mode::Insert {
-                    let mut keypress = keypress.clone();
-                    keypress.mods.set(Modifiers::SHIFT, false);
                     if let KeymapMatch::Full(command) =
-                        self.match_keymap(&[keypress], focus)
+                        self.match_keymap(std::slice::from_ref(&keypress), focus)
```
Comment (bugadani) [inline lapce-data/src/keypress/mod.rs:356]: Here I am either
 - introducing a bug
 - or I'm allowing key combinations like `shift+Delete` which I don't think were possible before

**Comment by bugadani** [comment]
I've cleaned up the line ending changes that inflated this PR.

**Comment by bugadani** [comment]
I have created a commit history that is hopefully easy to review. One downside is, that `keypress/mod.rs` still uses CRLF as line endings, to reduce the diff size.

Note that [Refactor keymap loader to avoid deserializing TOMLs twice](https://github.com/lapce/lapce/pull/191/commits/da13814138ca7a8a56228ca931dea60414a03510) may look like it has a weird code structure - impl blocks are generally mixed up. This was done to reduce the diff. The impl blocks are merged in 
[Split up into files](https://github.com/lapce/lapce/pull/191/commits/125794060f140e0ae0ef1941974dd5f3c2ea0dd0)

[Split up into files](https://github.com/lapce/lapce/pull/191/commits/125794060f140e0ae0ef1941974dd5f3c2ea0dd0) is basically a bunch of cut-paste operations, and fixing up the imports. The commit should not contain any code changes other than that, but don't trust me on this.

**Comment by bugadani** [comment]
Rebased

**Comment by dzhou121** [comment]
Thank you so much for the PR.
